redisvl/query/aggregate.py

```diff
@@ -0,0 +1,220 @@
+from typing import Any, Dict, List, Optional, Set, Tuple, Union
+
+import nltk
+from nltk.corpus import stopwords as nltk_stopwords
+from redis.commands.search.aggregation import AggregateRequest, Desc
+
+from redisvl.query.filter import FilterExpression
+from redisvl.redis.utils import array_to_buffer
+from redisvl.utils.token_escaper import TokenEscaper
+
+
+# base class
+class AggregationQuery(AggregateRequest):
+    """
+    Base class for aggregation queries used to create aggregation queries for Redis.
+    """
+
+    def __init__(self, query_string):
+        super().__init__(query_string)
+
+
+class HybridAggregationQuery(AggregationQuery):
+    """
+    HybridAggregationQuery combines text and vector search in Redis.
+    It allows you to perform a hybrid search using both text and vector similarity.
+    It scores documents based on a weighted combination of text and vector similarity.
+    """
+
+    DISTANCE_ID: str = "vector_distance"
+    VECTOR_PARAM: str = "vector"
+
+    def __init__(
+        self,
+        text: str,
+        text_field_name: str,
+        vector: Union[bytes, List[float]],
+        vector_field_name: str,
+        text_scorer: str = "BM25STD",
+        filter_expression: Optional[Union[str, FilterExpression]] = None,
+        alpha: float = 0.7,
+        dtype: str = "float32",
+        num_results: int = 10,
+        return_fields: Optional[List[str]] = None,
+        stopwords: Optional[Union[str, Set[str]]] = "english",
```

Comment:
I think if we list this `Optional` the default would be `None` generally speaking. So if we want it to have a default, then we can treat it like other args that have a default like `num_results` or `alpha`!